### PR TITLE
[build_base] Revert "[ci] Fix GPU docker builds"

### LINKED ISSUE #24392: [ci] Remove Docker GPU fix
Once NVIDIA updates their docker images, we can remove the fixes from https://github.com/ray-project/ray/pull/24336

### PR DESCRIPTION
**Pull request by krfricke**
Reverts ray-project/ray#24336

Closes #24392

### CODE REVIEW

File: docker/ray-ml/Dockerfile [hunk 1]
```diff
@@ -11,8 +11,8 @@ COPY requirements_tune.txt ./requirements_tune.txt
 COPY requirements_train.txt ./
 COPY requirements_upstream.txt ./
 
-RUN sudo apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/3bf863cc.pub
-RUN sudo apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64/7fa2af80.pub
+RUN if [[ "${PYTHON_VERSION}" =~ "3.6.*" ]]; then sudo apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64/3bf863cc.pub ; fi
+RUN if [[ "${PYTHON_VERSION}" =~ "3.6.*" ]]; then sudo apt-key adv --fetch-keys https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64/7fa2af80.pub ; fi
```
Comment (ddelange) [inline docker/ray-ml/Dockerfile:15]: just out of curiosity: why is the hotfix still needed for 3.6?

Comment (krfricke) [reply docker/ray-ml/Dockerfile:15]: It seems the base images have not been updated from nvidia

Comment (ddelange) [reply docker/ray-ml/Dockerfile:15]: doesn't the ray base image install python into a cuda ubuntu base image? i'm just wondering what makes py36 different from eg py37 🤔

judging from [here](https://github.com/ddelange/pycuda/runs/7510360317?check_suite_focus=true#step:7:533), the cuda base image has proper apt repos configured, and is ready to [install](https://github.com/ddelange/pycuda/blob/master/runtime/Dockerfile) python 36 (or any other) via apt-get and later install pytorch for cuda via pip

Comment (krfricke) [reply docker/ray-ml/Dockerfile:15]: Ah interesting - you might be right. For some reason it runs into problems only with the python 3.6 build

Comment (krfricke) [reply docker/ray-ml/Dockerfile:15]: But after the last commit it only fails in one of the two build jobs. I'm wondering if there is some caching going on that leads to passing in one case and failure in the other

Comment (ddelange) [reply docker/ray-ml/Dockerfile:15]: potentially a mismatch between ubuntu 20 and ubuntu 18:
https://github.com/ray-project/ray/blob/62820904015c3a1935721555d6914450ba2cb5ad/.buildkite/Dockerfile.gpu#L1
vs
https://github.com/ray-project/ray/blob/62820904015c3a1935721555d6914450ba2cb5ad/.buildkite/Dockerfile.gpu#L30-L31

but either way, the nvidia images should come with apt repos properly configured like i linked above.

also, i think python is not explicitly installed (`ARG PYTHON` is not consumed ^), but rather the nvidia `devel` image is used which will contain the `python3` apt package due to `software-properties-common` apt package, which by now probably resolves to python 3.9.

the `runtime` image (the one i [use](https://github.com/ddelange/pycuda/blob/70c9140084c0b88c0ae7472cc54c79ec0e6862d3/.github/workflows/build.yml#L13)) won't contain unwanted apt packages like python3 or development cuda libraries (used eg for building torch from source), and so you have an empty slate and you can [use](https://github.com/ddelange/pycuda/blob/70c9140084c0b88c0ae7472cc54c79ec0e6862d3/base/Dockerfile#L32-L33) deadsnakes to [install](https://github.com/ddelange/pycuda/blob/70c9140084c0b88c0ae7472cc54c79ec0e6862d3/runtime/Dockerfile) the python version you want

Comment (krfricke) [reply docker/ray-ml/Dockerfile:15]: That's a good point - we might revamp the docker image build process soon and we'll keep that in mind. We'll also deprecate 3.6 support soon so the problem should go away anyway

**Comment by krfricke** [comment]
We should update this PR after https://github.com/ray-project/ray/pull/28808 is merged

**Review by amogkam — APPROVED** [review]

**Pull request merged**